src/layers/scenegraph-layer/scenegraph-layer.js

```diff
@@ -0,0 +1,312 @@
+        ...super.visualChannels.size,
+        range: 'radiusRange',
+        property: 'radius',
+        channelScaleType: 'radius'
+      }
+    };
+  }
+
+  get layerInfoModal() {
+    return {
+      id: 'iconInfo',
+      template: this._layerInfoModal,
+      modalProps: {
+        title: 'How to draw icons'
+      }
+    };
+  }
+
+  static findDefaultLayerProps({fieldPairs, fields}) {
+    if (!fieldPairs.length) {
+      return [];
+    }
+
+    const iconFields = fields.filter(({name}) =>
+      name
+        .replace(/[_,.]+/g, ' ')
+        .trim()
+        .split(' ')
+        .some(seg => ({}.icon.some(t => t.includes(seg))))
+    );
+
+    if (!iconFields.length) {
+      return [];
+    }
+
+    // create icon layers for first point pair
+    const ptPair = fieldPairs[0];
+
+    const props = iconFields.map(iconField => ({
+      label: '???',
+      columns: {
+        lat: ptPair.pair.lat,
+        lng: ptPair.pair.lng
+      },
+      isVisible: true
+    }));
+
+    return props;
+  }
+
+  // TODO: fix complexity
+  /* eslint-disable complexity */
+  formatLayerData(_, allData, filteredIndex, oldLayerData, opt = {}) {
+    const {
+      colorScale,
+      colorDomain,
+      colorField,
+      color,
+      columns,
+      sizeField,
+      sizeScale,
+      sizeDomain,
+      visConfig: {
+        radiusRange,
+        colorRange,
+        sizeScale: sizeScale3D,
+        angleX = 0,
+        angleY = 0,
+        angleZ = 0
+      }
+    } = this.config;
+
+    // point color
+    const cScale =
+      colorField &&
+      this.getVisChannelScale(
+        colorScale,
+        colorDomain,
+        colorRange.colors.map(hexToRgb)
+      );
+
+    // point radius
+    const rScale =
+      sizeField && this.getVisChannelScale(sizeScale, sizeDomain, radiusRange);
+
+    const getPosition = this.getPosition(columns);
+
+    if (!oldLayerData || oldLayerData.getPosition !== getPosition) {
+      this.updateLayerMeta(allData, getPosition);
+    }
+
+    let data;
+    if (
+      oldLayerData &&
+      oldLayerData.data &&
+      opt.sameData &&
+      oldLayerData.getPosition === getPosition
+    ) {
+      data = oldLayerData.data;
+    } else {
+      data = filteredIndex.reduce((accu, index) => {
+        const pos = getPosition({data: allData[index]});
+
+        // if doesn't have point lat or lng, do not add the point
+        // deck.gl can't handle position = null
+        if (!pos.every(Number.isFinite)) {
+          return accu;
+        }
+
+        accu.push({
+          index,
+          data: allData[index]
+        });
+
+        return accu;
+      }, []);
+    }
+
+    const getRadius = rScale
+      ? d => this.getEncodedChannelValue(rScale, d.data, sizeField)
+      : 1;
+
+    const getColor = cScale
+      ? d => this.getEncodedChannelValue(cScale, d.data, colorField)
+      : color;
+
+    return {
+      data,
+      getPosition,
+      getColor,
+      getRadius,
+      sizeScale: sizeScale3D,
```

Comment:
basically, you only need to return `data` and `getPosition` here. `sizeScale` and `getOrientation` can be assigned in the `renderLayer` function below